tests/_files/TestIteratorAggregate2.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/* This class is used for testing a chain of IteratorAggregate objects, since
+ * PHP does allow IteratorAggregate::getIterator() to return an instance of the
+ * same class. */
```

Comment:
See: [`zend_user_it_get_new_iterator()`](https://github.com/php/php-src/blob/php-7.1.5/Zend/zend_interfaces.c#L285). If `IteratorAggregate::getIterator()` returns an instance of the same class, PHP throws an exception. This is a basic attempt to prevent infinite recursion, although it is certainly still possible if two different IteratorAggregate implementations return instances of each other.